ibis/backends/duckdb/__init__.py

```diff
@@ -67,6 +112,24 @@ def do_connect(
         )
         self._meta = sa.MetaData(bind=self.con)
 
+    def register(
+        self,
+        file_name: str | Path,
+        table_name: str | None = None,
+    ) -> None:
+        """Register an external file (csv or parquet) as a table in the current
+        connection database
+
+        Parameters
+        ----------
+        file_name
+            Name of the parquet or CSV file
+        table_name
+            Name for the created table.  Defaults to filename if not given
```

Comment:
Might be worth documenting here the behavior of `table_name` with respect to dashes if `table_name` is `None`.